reactive/kubernetes_master.py

```diff
@@ -1361,30 +1367,30 @@ def remove_rbac_resources():
 @when('nrpe-external-master.available')
 @when_any('config.changed.nagios_context',
           'config.changed.nagios_servicegroups')
-def update_nrpe_config(unused=None):
-    services = (
-        'snap.kube-apiserver.daemon',
-        'snap.kube-controller-manager.daemon',
-        'snap.kube-scheduler.daemon'
-    )
+def update_nrpe_config():
+    services = ['snap.{}.daemon'.format(s) for s in master_services]
+
+    plugin = install_nagios_plugin_from_file('templates/nagios_plugin.py',
+                                             'check_k8s_master.py')
     hostname = nrpe.get_nagios_hostname()
     current_unit = nrpe.get_nagios_unit_name()
     nrpe_setup = nrpe.NRPE(hostname=hostname)
     nrpe.add_init_service_checks(nrpe_setup, services, current_unit)
+    nrpe_setup.add_check('k8s-api-server',
+                         'Verify that the Kubernetes API server is accessible',
+                         str(plugin))
     nrpe_setup.write()
 
 
 @when_not('nrpe-external-master.available')
 @when('nrpe-external-master.initial-config')
-def remove_nrpe_config(nagios=None):
+def remove_nrpe_config():
     remove_state('nrpe-external-master.initial-config')
```

Comment:
let's remove this state at the end.  otherwise, we could fail in this handler and incorrectly assume we've cleaned up.

Reply:
yes, I'd love to change existing code on a merged PR. :)

That said, this seems reasonable, want me to make a new PR for that?

Reply:
Done in #65